packages/neos-ui/src/index.js

```diff
@@ -152,6 +152,30 @@ function * application() {
     });
 
     fetchWithErrorHandling.registerGeneralErrorHandler((message = 'unknown error') => {
+        // Check if the message is a Flow exception response
+        if (message.indexOf('Flow-Debug-Exception-Header') >= 0) {
+            const htmlContainer = document.createElement('div');
+            htmlContainer.innerHTML = message;
+            const exceptionHeader = htmlContainer.querySelector('.Flow-Debug-Exception-Header');
+            if (exceptionHeader && exceptionHeader.textContent) {
+                const exceptionSubject = exceptionHeader.querySelector('.ExceptionSubject');
+                message = exceptionSubject.textContent + ' - Check the logs for details';
+            } else {
+                message = 'Unknown error from unexpected HTML response. Check the logs for details.';
+            }
+        }
+        // Check if the message is a JSON string
+        else if (message.indexOf('{') >= 0) {
```

Comment:
Why have this condition at all here? Why not just `else {`?
Also if you insist on having it, you could at least harden it to this: `message.trim()[0] === '{'` (every JSON begins with "{", right?).
But I would not do it. The best way to detect if it's JSON or not is to actually parse it.

Reply:
I prefer to keep the condition to skip the parse when not needed. But I can adjust it as you said.

Reply:
Alright, so you are trying to optimise performance of it? I guess it should almost be identical, as the parser would tumble upon any first invalid character, but I don't mind.